Making the pit implementation more complex

y-arjun-y opened the pull request:
I've added fluctuating commodity scores and went for the bull & bear variant of the game: with a 20% increase and 20% decrease respectively. This should hopefully add more depth to the game, but I am open to adding more implementation if needed, I have some more ideas:

1. Defining more kwargs specifically for Pit to introduce different trading strategies.
2. Enabling the agents to interact with each other in some primitive way.
3. Allowing a primitive form of market analysis for both agents which can inform their decisions.

acostarelli commented:
When I attempt to run this with two random agents, I get an error.
```
pit.py", line 220, in play
    agent.send_message(other_agent.agent_id, message_content, message_type)
AttributeError: 'RandomAgent' object has no attribute 'send_message'
```

acostarelli commented:
[ ] I see that both agents are on the same team based on their instantiation. Is this intentional?
[ ] Why is `volume_traded` unusued?
[ ] Why are messages generated between agents random?
[ ] Why must agents send a trade inquiry every turn? Shouldn't this be a choice?
[ ] Why is it random if an agent is interested in a trade offer?
[ ] Why are agents offered quantities in trade offers random?
[ ] Why are agents decisions regarding trade offers random?
[ ] Why are random price fluctuations and random bear/bull market effects applied for direct trades?
[ ] Due to all the randomness, I don't believe this message capability adds any strategy complexity at all. It just adds more randomness.
[ ] When a trade offer is accepted, an agent just takes it off the market commodities. There is no interaction between the two agents. Why is it not that the commodities from one agent are transferred to the other -- it is a trade, right?

y-arjun-y commented:
[X] I see that both agents are on the same team based on their instantiation. Is this intentional? - fixed, was not intentional
[X] Why is `volume_traded` unused?  - fixed, the `calculate_market_trends_adjust_values` function now takes into account the `volume_traded` as well
[X] Why must agents send a trade inquiry every turn? Shouldn't this be a choice? - made it a choice now whether to inquire or directly make an offer **that is a part of the available actions**
[X] When a trade offer is accepted, an agent just takes it off the market commodities. There is no interaction between the two agents. Why is it not that the commodities from one agent are transferred to the other -- it is a trade, right? - correct, I've updated the logic so that the trade occurs properly

## About randomness
I'm not exactly sure how to go about making the choices for communication-based on the agent's choice (using, I suspect, the `available_actions` dictionary), I would appreciate any pointers here, I do agree that as it stands the messaging feature doesn't add much more complexity (even though I do feel it is closer to the more complex implementations of Pit I've discussed before) - I've committed two versions - one with the messaging implementation and one without.
